index.js

```diff
@@ -379,3 +379,112 @@ exports.requestMultiItemBasic = (itemIds, cb) => {
       })
   }
 }
+
+
+// Just like auth except in case of a successful call, returns a promise that resolves to the value of the
+// callback
+
+exports.promiseAuth = (cb) => {
+  if (!exports.credsKey || !exports.credsSecret || !exports.credsBase) {
+    console.error('No credentials set')
+    if (cb) cb('No credentials set', false)
+  } else {
+    // uses the basic auth method to ask for the token
+    return new Promise((resolve, reject) => {
+      request.post(exports.credsBase + 'token', {
+        'auth': {
+          'user': exports.credsKey,
+          'pass': exports.credsSecret
+        }
+      },
+      (error, response, body) => {
+        if (error) console.error(error)
```

Comment:
It seems to me that this would also require a callback to be invoked here. I noticed it doesn't already do that but not sure when that specific case will occur and if we want any info regarding that case.

Reply:
Yeah I am not sure what would cause this but I am a little reticent to change it since I don't know what depends on this behavior